cpp/dolfinx/fem/Form.h

```diff
@@ -359,8 +359,42 @@ class Form
         break;
       }
       case IntegralType::exterior_facet:
-        // Exterior and interior facets are treated the same
-        [[fallthrough]];
+      {
+        // Get the codimension of the mesh
+        const int tdim = _mesh->topology()->dim();
+        const int codim = tdim - mesh.topology()->dim();
+        assert(codim >= 0);
+
+        if (codim == 0)
```

Comment:
Should ` mapped_entities.reserve(...)` be moved inside the case? Looks like the reserved length should be doubled for exterior facets.

Reply:
No; `mapped_entities.reserve(entities.size())` is correct because `entities` contains (flattened) `(cell, local_facet)` pairs for exterior facet integrals.